### PR TITLE
Compress NonMatching::MappingInfo for faces

### PR DESCRIPTION
**Pull request by bergbauer**
This PR enables compression for faces on affine grids speeding up evaluation significantly for low polynomial degrees.

@kronbichler @peterrum

### CODE REVIEW

File: include/deal.II/non_matching/mapping_info.h [hunk 1]
```diff
@@ -1284,8 +1284,12 @@ namespace NonMatching
     resize_unit_points(n_unit_points);
     resize_data_fields(n_data_points);
 
-    MappingData mapping_data;
-    cell_index = 0;
+    MappingData  mapping_data;
+    MappingData  mapping_data_last_cell;
```
Comment (kronbichler) [inline include/deal.II/non_matching/mapping_info.h:1288]: Do you mean
```suggestion
    MappingData  mapping_data_previous_cell;
```
(It is not wrong, but I think 'previous' is not ambiguous.)

**Review by kronbichler — APPROVED** [review]
I think this looks good, thank you!

**Review by kronbichler — APPROVED** [review]

**Comment by bergbauer** [comment]
The failure is unrelated, should we trigger the checks again?

**Review by peterrum — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Pull request merged**